packages/models/src/actor/registry.ts

```diff
@@ -1,41 +1,70 @@
 /**
  * @tutorial https://github.com/ckb-js/kuai/issues/17#issuecomment-1305896619
- *
- * The registry is used as an IoC of actors
- * At this early stage, a custom registry is used for demostration, will be replaced by inversify
  */
+import fs from 'node:fs'
+import { resolve } from 'node:path'
 
-import { ActorRef, ActorURI } from './interface'
-import type { Actor, ActorConstructor } from './actor'
+import { Container } from 'inversify'
+import type { ActorRef, ActorURI } from './interface'
+import type { Actor } from './actor'
+import { DuplicatedActorException, InvalidActorURIException, ProviderKey } from '../utils'
 
 export class Registry {
-  #actors: Map<ActorURI, Actor> = new Map()
-  #actorConstructor: ActorConstructor
+  #actors: Set<ActorURI> = new Set()
+  #container: Container = new Container()
 
-  constructor(actorConstructor: ActorConstructor) {
-    this.#actorConstructor = actorConstructor
-  }
-
-  spawn = ({ parent, name }: Partial<{ parent: ActorRef; name: string | symbol }>): ActorRef => {
-    // TODO: use IoC to spawn an actor
-    const actor = new this.#actorConstructor(parent, name)
-    const ref = actor.ref
-    this.#actors.set(ref.uri, actor)
-    return ref
+  isLive = (uri: ActorURI): boolean => {
+    return this.#actors.has(uri)
   }
 
   find = (uri: ActorURI): Actor | undefined => {
-    if (this.isLive(uri)) {
-      return this.#actors.get(uri)
+    try {
+      return this.#container.get(uri)
+    } catch {
+      return undefined
     }
-    return undefined
   }
 
   list = (): IterableIterator<ActorURI> => {
     return this.#actors.keys()
   }
 
-  isLive = (uri: ActorURI): boolean => {
-    return this.#actors.has(uri)
+  // TODO: this method is not tested
+  load = (path: string): void => {
```

Comment:
How about using `@Bind`, but not scanning the directory?

Reply:
How to detect all bound interfaces on launch without scanning the project

Reply:
May be like this?
```
const register = new Registry()
function Bind(target: new (...args: Array<unknown>) => unknown) {
  register.bind(target)
}

@Bind
class ChildActor {

}
```

Reply:
> May be like this?
> 
> ```
> const register = new Registry()
> function Bind(target: new (...args: Array<unknown>) => unknown) {
>   register.bind(target)
> }
> 
> @Bind
> class ChildActor {
> 
> }
> ```

The decorator works on instantiation so `register.bind(target)` will be evaluated on instantiation of `ChildActor`, but our goal is to instantiate `ChildActor` by the container/register. It turns into a chicken and egg problem.

Reply:
I see this conversation has been marked as resolved, but this PR is not approved, are there any pending reviews not submitted?